lib/pbench/cli/agent/commands/tools/clear.py

```diff
@@ -26,16 +26,20 @@ def __init__(self, context):
     def execute(self) -> int:
         errors = 0
 
-        groups = [group for group in self.context.group.split(",") if group]
+        groups = self.context.group.split(",")
         # groups is never empty and if the user doesn't specify a --group,
         # then it has the value "default"
         for group in groups:
-            if self.verify_tool_group(group) != 0:
+            if not group:
+                self.logger.warn("Blank group name specified; skipping.")
```

Comment:
Strictly speaking, _failing_ isn't "skipping" ... and the `errors = 1` prevents a script from reliably determining that anything succeeded, so we might as well have just terminated here. In fact, that would be safer, as the failure status should suggest to the script/user that nothing was changed.

Reply:
> Strictly speaking, _failing_ isn't "skipping" ... and the `errors = 1` prevents a script from reliably determining that anything succeeded, so we might as well have just terminated here. In fact, that would be safer, as the failure status should suggest to the script/user that nothing was changed.

Well, nothing changed only for the groups that came after the list encounters its first empty element, however, by this point we have already removed groups that came before an empty element. If we were to terminate on empty element without having any side effects we should pre-check the list for an empty element and terminate if we encounter empty element.

Reply:
> the failure status should suggest to the script/user that nothing was changed.

Not to quibble or anything, but if the user requests that multiple things be cleared, and if the first one gets cleared, and then if we discover that a subsequent one doesn't exist, we'll (eventually) report an error, even though something _was_ changed.

Ideally, we would validate the format of our input, then validate the existence of all the targets, and then clear them, so that we don't **_do_** anything if we cannot do everything...but, that's a lot more work, for not a lot of gain, and it's still subject to problems if something else concurrently changes the environment.

Still, as I mentioned elsewhere, for the blanks at least, it's pretty easy to validate the input before proceeding, so I propose that we do that.